lib/Model/Importer/DebugInfo/DwarfImporter.cpp

```diff
@@ -1057,17 +926,16 @@ class DwarfToModelConverter : public BinaryImporterHelper {
     resolveAllTypes();
     T.advance("Create model functions", true);
     createFunctions();
-    T.advance("Type system cleanup", true);
-    cleanupTypeSystem();
-    T.advance("Apply fixes to the model", true);
-    fixModel(Model);
+    T.advance("Remove types that depend on unresolved placeholders", true);
+    purgeUnresolvedPlaceholders();
+    T.advance("Remove types that couldn't be imported fully", true);
+    purgeInvalidTypes(Model);
```

Comment:
This is nice to have, it makes our importers more robust, however we might miss on some invalid things we generate.
I'm prone to keep this, but can you give me more context?
Is this a "new feature" or did you need to introduce this for some reason?

Reply:
The context is that we had three distinct versions of this logic, with varying degrees of complexity. Instead of doing the dance and maintaining all that, I just said screw it and added a dedicated helper for all of them to reuse.

Reply:
I'm not sure I understand. What are the three distinct versions of this logic? Can you point me to them?

Reply:
> Can you point me to them?

First of all straightforward type verification, since it's very similar: it fails after the first problem, but the logic is similar.

Then there were distinct versions in the dwarf importer and the PDB importer.

TL;DR: this is just a verification plus some softening passes on top (for example, allowing only a field of a struct/union to be dropped instead of the entire type and everything that depends on it).